metals/src/main/scala/scala/meta/internal/metals/debug/ExtendedScalaMainClass.scala

```diff
@@ -8,7 +8,25 @@ import scala.meta.io.AbsolutePath
 import ch.epfl.scala.bsp4j.ScalaMainClass
 import ch.epfl.scala.{bsp4j => b}
 
-case class ExtendedScalaMainClass(
+/**
+ * Wrapper around the bsp4j.ScalaMainClass to provide additional information which may be used by client.
+ *
+ * For backward compatibility reasons, it provides the same fields as the bsp4j.ScalaMainClass,
+ * so it's safe for older clients to work without any change:
+ * @param class the fully qualified name of the class
+ * @param arguments the arguments to pass to the main method
+ * @param jvmOptions the jvm options to pass to the jvm
+ * @param environmentVariables the environment variables to pass to the process
+ *
+ * However, it also provides two additional fields:
+ * @param kind allows client to distinguish between old, bsp4j.ScalaMainClass and new ExtendedScalaMainClass
+ * @param shellCommand which is the command to run in the shell to start the main class
+ * ---
+ * To sum up:
+ * - allow old clients to work without any change
```

Comment:
Old commands should work without any changes anyway, why would adding a new field cause any issues?

Reply:
Isn't that what I wrote? 😆 I meant that this change is backward compatible

Reply:
Ach, ok wasn't sure about it since you wrote above:
```
My idea is following:

extend in backward compatible way run lenses. Allow old clients to use them as before, but at the same time allow new clients to take advantage of shellCommand
```

I thought you meant we should extend in a compatible way and we don't do it yet.